router/handler_ctrl/route.go

```diff
@@ -156,13 +162,27 @@ func (rh *routeHandler) connectEgress(msg *channel.Message, attempt int, ch chan
 				if peerData, err := dialer.Dial(route.Egress.Destination, circuitId, xgress.Address(route.Egress.Address), bindHandler, ctx); err == nil {
 					rh.success(msg, attempt, route, peerData, log)
 				} else {
-					rh.fail(msg, attempt, route, errors.Wrapf(err, "error creating route for [c/%s]", route.CircuitId), log)
+					var headerError *byte
+					if terr, ok := err.(ctrl_pb.InvalidTerminatorError); ok {
+						var errCode byte = ctrl_msg.ErrorTypeInvalidTerminator
+						headerError = &errCode
+						if oerr, ok := terr.InnerError.(*net.OpError); ok {
+							switch oerr.Err {
+							case syscall.ECONNREFUSED:
+								errCode = ctrl_msg.ErrorTypeConnectionRefused
+							case syscall.ETIMEDOUT:
+								errCode = ctrl_msg.ErrorTypeDialTimedOut
+							}
+						}
+						headerError = &errCode
+					}
+					rh.fail(msg, attempt, route, errors.Wrapf(err, "error creating route for [c/%s]", route.CircuitId), headerError, log)
 				}
 			} else {
```

Comment:
Should return misconfigured terminator if the binding is invalid or the dialer can't be created